### DISCUSSION TITLE
Symfony 6 support

### DISCUSSION THREAD

**Discussion by Nikoblood** [overblog/GraphQLBundle#932] [Q&A] [answered]
Hello there! Do you planning to add support for Symfony 6 in the nearest future?

**Comment by ruudk** [comment]
Of course 😊 could you give it a try and create a PR? :) 

I think the first step would be to enable 5.4 here https://github.com/overblog/GraphQLBundle/blob/master/.github/workflows/ci.yml#L25
and see if those tests pass...

If there are deprecations, you can fix them easily.

Then, in another PR, you can add 6.0 support via the same way + updating the composer.json file.

Not sure if you should target 0.14 branch or master, but that's up to @mcg-web

**Comment by Nikoblood** [comment]
Thanks, I will try! We are using your bundle in our projects, and the last project is already updated to symfony 5.4, using last stable version of `overblog/graphql-bundle`: `v0.13.6`. 

I`ll start from PR, related to symfony 5.4

**Reply by ruudk** [reply]
Yeah, it works because we don't exclude 5.4. But it's good that we start testing with that release to spot bugs. 

Thanks, let me know if you need further help 💪

**Comment by Tib-z** [comment, marked as answer]
Hi, 
As symfony 5.3 is already EOL, I plan to use symfony 6 for new projects (at least 5.4, but would like 6.0).
@Nikoblood do you have any update regarding symfony 6.0 (or 5.4) migration/compatibility progress?
@ruudk I saw your symfony 6 commit is already in the master branch. 
https://github.com/overblog/GraphQLBundle/commit/e3b83428af6ff32dde889176aedcc7b9c56e5acf
Any plans on sf 6 support on a stable branch? I'm not sure about the roadmap.. Or do you think we should stick to sf 5.4 and use 0.13? However according to Symfony's backward compatibility promise as long as deprecations are fixed on 5.4, then 6.0 the next major will be compatible as well. @Nikoblood have you faced any issues with 0.13 on Sf 5.4?

**Reply by calvera** [reply]
i use symfony 5.4 with version 0.14.1 of the package

**Reply by Nikoblood** [reply]
@Carobit Symfony 6 has BC breaks, that needs to be done in overblog/GraphQLBundle. Because of this, I assume, that there must be new major version of this bundle, that will not have backward compatibility for older packages / versions of symfony.

Regarding to that, I think that there are too big responsibility for me to update this package for Symfony 6 and this must be done by maintainers of the package.

**Reply by Nikoblood** [reply]
@Carobit Also I must to say, that develop branch of this bundle is already supports Symfony 6. So our project has dependecy on dev-master ( we are using one concrete stable commit that has green tests ).

**Reply by ruudk** [reply]
Master branch is using https://github.com/webonyx/graphql-php dev-master and is waiting for a tagged release. 

See:
- https://github.com/webonyx/graphql-php/issues/1037

**Reply by ruudk** [reply]
That being said, we are using dev-master of both projects in production and it works great. We pin to a specific commit hash so that we don't accidentally break things.

**Reply by Tib-z** [reply]
Good to know, thanks for the heads up.
